operator/es_client.go

```diff
@@ -312,9 +352,37 @@ func (c *ESClient) waitForEmptyEsNode(ctx context.Context, pod *v1.Pod) error {
 			},
 		).R().
 		Get(c.Endpoint.String() + "/_cat/shards?h=index,ip&format=json")
+
 	if err != nil {
 		return err
 	}
+
+	if !resp.IsSuccess() {
+		return fmt.Errorf("HTTP endpoint responded with not expected status code %d", resp.StatusCode())
+	}
+
+	// make sure the IP is still excluded, this could have been updated in the meantime.
+	if err = c.excludePodIP(pod); err != nil {
```

Comment:
Why should the IP not be in the exclude-list at this point?

Reply:
I thought it over again and looks like it's a mistake.
It doesn't make sense to check if pod IP is in the excluded list after we finish waiting for condition (migration all shards). So I removed this part.